ClassifierDLModel fails if empty string is passed as input

<!--- Provide a general summary of the issue in the Title above -->
ClassifierDLModel fails if empty string is passes as input 

## Description
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->

## Expected Behavior
df.show(3) works and should ignore empty embedding arrays
## Current Behavior
<!--- Tell us what happens instead of the expected behavior -->

Caused by: java.lang.IllegalArgumentException: cannot create Tensors with a 0 dimension
	at org.tensorflow.Tensor.fillShape(Tensor.java:750)
	at org.tensorflow.Tensor.create(Tensor.java:142)
	at org.tensorflow.Tensor.create(Tensor.java:126)
	at com.johnsnowlabs.ml.tensorflow.TensorResources.createTensor(TensorResources.scala:22)
	at com.johnsnowlabs.ml.tensorflow.TensorflowClassifier.predict(TensorflowClassifier.scala:134)
	at com.johnsnowlabs.nlp.annotators.classifier.dl.ClassifierDLModel.annotate(ClassifierDLModel.scala:150)
	at com.johnsnowlabs.nlp.AnnotatorModel$$anonfun$dfAnnotate$1.apply(AnnotatorModel.scala:35)
	at com.johnsnowlabs.nlp.AnnotatorModel$$anonfun$dfAnnotate$1.apply(AnnotatorModel.scala:34)
	at org.apache.spark.sql.catalyst.expressions.ScalaUDF$$anonfun$2.apply(ScalaUDF.scala:109)
	at org.apache.spark.sql.catalyst.expressions.ScalaUDF$$anonfun$2.apply(ScalaUDF.scala:108)
	at org.apache.spark.sql.catalyst.expressions.ScalaUDF.eval(ScalaUDF.scala:1064)
	... 107 more
## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
Ignore if embeddings are empty arrays

## Steps to Reproduce
<!--- Provide a link to a live example or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
```
val spark: SparkSession = SparkSession.builder()
      .appName("spark-nlp-app")
      .master("local[*]")
      .config("spark.driver.host", "127.0.0.1")
      .config("spark.driver.memory", "22G")
      .config("spark.executor.memory", "22G")
      .config("spark.driver.maxResultSize", "0")
      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
      .config("spark.kryoserializer.buffer.max", "1000m")
      .config("spark.ui.enabled", "false")
      .getOrCreate()

    val documentAssembler = new DocumentAssembler()
      .setInputCol("text")
      .setOutputCol("document")

   val use = UniversalSentenceEncoder.pretrained(name = "tfhub_use", lang = "en")
      .setInputCols("document")
      .setOutputCol("sentence_embeddings")

    val sarcasmDL = ClassifierDLModel.pretrained(name = "classifierdl_use_sarcasm")
      .setInputCols("sentence_embeddings")
      .setOutputCol("sarcasm")

 val nlpPipeline = new Pipeline().setStages(
      Array(
        documentAssembler,
        use,
        sarcasmDL ))
val data = Seq((1, ""),
      (2, "We have Amazingly protective Police while the arsonist is running away and the Police car like it's not his Job. Holly crap. Whom do we rely right nowCrying face\nFlag of Myanmar WE NEED TO BE SAVED."))

    val testData1 = spark.createDataFrame(data).toDF("id", "text")

    val prediction = nlpPipeline.fit(testData).transform(testData)
    prediction.show(10)
```

## Context
<!--- How has this bug affected you? What were you trying to accomplish? -->
I tried to introduce pre-processing and remove hashtags and mentions, some tweets had only hashtags so the result was a empty string
## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Spark NLP version: "2.4.7"
* Apache NLP version: "2.7.3"
* Java version (java -version): 1.8_275 openJDK
* Operating System and version: MacOS

--- a/src/main/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/ClassifierDLModel.scala
+++ b/src/main/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/ClassifierDLModel.scala
@@ -147,7 +147,10 @@ class ClassifierDLModel(override val uid: String)
       .toSeq
       .sortBy(_._1)
 
-    getModelIfNotSet.predict(sentences, getConfigProtoBytes)
+    if(sentences.nonEmpty)
+      getModelIfNotSet.predict(sentences, getConfigProtoBytes)
+    else Seq.empty[Annotation]
+
   }
 
   override def onWrite(path: String, spark: SparkSession): Unit = {

--- a/src/main/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/SentimentDLModel.scala
+++ b/src/main/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/SentimentDLModel.scala
@@ -105,7 +105,10 @@ class SentimentDLModel(override val uid: String)
       .toSeq
       .sortBy(_._1)
 
-    getModelIfNotSet.predict(sentences, getConfigProtoBytes, $(threshold), $(thresholdLabel))
+    if(sentences.nonEmpty)
+      getModelIfNotSet.predict(sentences, getConfigProtoBytes, $(threshold), $(thresholdLabel))
+    else Seq.empty[Annotation]
+
   }
 
   override def onWrite(path: String, spark: SparkSession): Unit = {

--- a/src/test/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/ClassifierDLTestSpec.scala
+++ b/src/test/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/ClassifierDLTestSpec.scala
@@ -48,6 +48,47 @@ class ClassifierDLTestSpec extends FlatSpec {
 
   }
 
+  "ClassifierDL" should "not fail on empty inputs" taggedAs SlowTest in {
+
+    val testData = ResourceHelper.spark.createDataFrame(Seq(
+      (1, "This is my first sentence. This is my second."),
+      (2, "This is my third sentence. . . . .... ..."),
+      (3, "")
+    )).toDF("id", "text")
+
+    val documentAssembler = new DocumentAssembler()
+      .setInputCol("text")
+      .setOutputCol("document")
+
+    val sentence = new SentenceDetector()
+      .setInputCols("document")
+      .setOutputCol("sentence")
+
+    val useEmbeddings = UniversalSentenceEncoder.pretrained()
+      .setInputCols("document")
+      .setOutputCol("sentence_embeddings")
+
+    val sarcasmDL = ClassifierDLModel.pretrained(name = "classifierdl_use_sarcasm")
+      .setInputCols("sentence_embeddings")
+      .setOutputCol("sarcasm")
+
+    val pipeline = new RecursivePipeline()
+      .setStages(Array(
+        documentAssembler,
+        sentence,
+        useEmbeddings,
+        sarcasmDL
+      ))
+
+    val pipelineDF = pipeline.fit(testData).transform(testData)
+    pipelineDF.select("sentence.result").show(false)
+    pipelineDF.select("sentence_embeddings.result").show(false)
+    pipelineDF.select("sarcasm.result").show(false)
+
+    pipelineDF.show()
+
+  }
+
   "ClassifierDL" should "correctly download and load pre-trained model" taggedAs FastTest in {
     val classifierDL = ClassifierDLModel.pretrained("classifierdl_use_trec50")
     classifierDL.getClasses.foreach(x=>print(x+", "))

--- a/src/test/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/SentimentDLTestSpec.scala
+++ b/src/test/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/SentimentDLTestSpec.scala
@@ -52,4 +52,46 @@ class SentimentDLTestSpec extends FlatSpec {
 
   }
 
+  "SentimentDL" should "not fail on empty inputs" taggedAs SlowTest in {
+
+    val testData = ResourceHelper.spark.createDataFrame(Seq(
+      (1, "This is my first sentence. This is my second."),
+      (2, "This is my third sentence. . . . .... ..."),
+      (3, "")
+    )).toDF("id", "text")
+
+    val documentAssembler = new DocumentAssembler()
+      .setInputCol("text")
+      .setOutputCol("document")
+
+    val sentence = new SentenceDetector()
+      .setInputCols("document")
+      .setOutputCol("sentence")
+
+    val useEmbeddings = UniversalSentenceEncoder.pretrained()
+      .setInputCols("document")
+      .setOutputCol("sentence_embeddings")
+
+    val sentiment = SentimentDLModel.pretrained(name = "sentimentdl_use_twitter")
+      .setInputCols("sentence_embeddings")
+      .setThreshold(0.7F)
+      .setThresholdLabel("neutral")
+      .setOutputCol("sentiment")
+
+    val pipeline = new RecursivePipeline()
+      .setStages(Array(
+        documentAssembler,
+        sentence,
+        useEmbeddings,
+        sentiment
+      ))
+
+    val pipelineDF = pipeline.fit(testData).transform(testData)
+    pipelineDF.select("sentence.result").show(false)
+    pipelineDF.select("sentence_embeddings.result").show(false)
+    pipelineDF.select("sentiment.result").show(false)
+
+    pipelineDF.show()
+
+  }
 }

